thefuck/shells.py

```diff
@@ -153,16 +153,16 @@ def _get_history_line(self, command_script):
     'bash': Bash(),
     'fish': Fish(),
     'zsh': Zsh(),
-    '-csh': Tcsh(),
+    'csh': Tcsh(),
     'tcsh': Tcsh()})
 
 
 def _get_shell():
     try:
-        shell = Process(os.getpid()).parent().cmdline()[0]
+        shell = Process(os.getpid()).parent().name()
     except TypeError:
-        shell = Process(os.getpid()).parent.cmdline[0]
-    return shells[os.path.basename(shell)]
+        shell = Process(os.getpid()).parent.name()
```

Comment:
BTW, shouldn't this line be `Process(os.getpid()).name()` if parent is None?

Since the version of psutil is not specified I don't know if I should change this line.

[Process.parent](https://pythonhosted.org/psutil/_modules/psutil.html#Process.parent)